src/ape/managers/chain.py

```diff
@@ -359,6 +363,73 @@ def revert_to_block(self, block_number: int):
         }
 
 
+class ContractCache(BaseManager):
```

Comment:
Can it subclass `dict`?

Then will have these methods:
```py
ct = chain.contracts[addr]
ct = chain.contract.get(addr, None)
```
etc.

could maybe even have `chain.contracts[addr] = ct`

Reply:
hmm, yes but still have to have custom definitions for them because this is not a static list